fabric/patches/2_0002-Add-FPC-SCCs-TLCC-via-SCC-Extension-point.patch

```diff
@@ -0,0 +1,49 @@
+From 583806cbadeff089560b88aa920bf978f55abd17 Mon Sep 17 00:00:00 2001
+From: michael steiner <michael.steiner@intel.com>
+Date: Fri, 31 Jul 2020 14:23:25 -0700
+Subject: [PATCH 2/2] Add FPC SCCs (TLCC) via SCC Extension point
+
+Signed-off-by: michael steiner <michael.steiner@intel.com>
+---
+ internal/peer/node/sccs_opt_fpc.go | 29 +++++++++++++++++++++++++++++
+ 1 file changed, 29 insertions(+)
+ create mode 100644 internal/peer/node/sccs_opt_fpc.go
+
+diff --git a/internal/peer/node/sccs_opt_fpc.go b/internal/peer/node/sccs_opt_fpc.go
+new file mode 100644
+index 000000000..0e297f0e6
+--- /dev/null
++++ b/internal/peer/node/sccs_opt_fpc.go
+@@ -0,0 +1,29 @@
++//+build FPC
++
++/*
++ Copyright 2020 Intel Corporation
++
++SPDX-License-Identifier: Apache-2.0
++*/
++
++package node
++
++import (
++	"github.com/hyperledger-labs/fabric-private-chaincode/tlcc"
++	"github.com/hyperledger/fabric/core/aclmgmt"
++	"github.com/hyperledger/fabric/core/peer"
++	"github.com/hyperledger/fabric/core/scc"
++)
++
++type tlccFactory struct{}
++
++func (f tlccFactory) newOptSCC(aclProvider aclmgmt.ACLProvider, p *peer.Peer) scc.SelfDescribingSysCC {
++	return tlcc.New(p)
++}
```

Comment:
Just wondering if we could replace this factory object with just a factory function, similar how http.handlers are used. The SSCRegistry would keep a list of factory functions instead of objects. This way we prevent any naming conflicts in the `node` package with other customSSC.

sccs_opt_fpc.go becomes:
```go
func init() {
    logger.Debug("Enabling tlcc-factory")
    optionalSCCs.add(func(aclProvider aclmgmt.ACLProvider, p *peer.Peer) scc.SelfDescribingSysCC {
        return tlcc.New(p)
    })
}
```

ssc_opt.go becomes:
```go
type FactoryFunc func(aclProvider aclmgmt.ACLProvider, p *peer.Peer) scc.SelfDescribingSysCC

type optionalSCCRegistry struct {
	registeredFactories []FactoryFunc
}

func (f *optionalSCCRegistry) add(ff FactoryFunc) {
	f.registeredFactories = append(f.registeredFactories, ff)
}

func (f *optionalSCCRegistry) list() []FactoryFunc {
	return f.registeredFactories
}
```
and in start.go
```go
for _, f := range optionalSCCs.list() {
		osi := f(aclProvider, peerInstance)
...
```

This is just an alternative implementation proposal. Not sure if this is actually better or not, but reduces the need to have a factory object for each custom scc and just provides a factory function. Maybe the real factory is a bit more future proof, in particular, when the factory would set some default values, but in this case we would introduce a factory for the factory. Let's keep it simple.

Reply:
didn't realize that you can just pass a function to an interface having a single function and don't need a type in that case. Definitely much better, will change!